providers/implementations/exchange/dh_exch.c

```diff
@@ -345,6 +345,8 @@ static int dh_set_ctx_params(void *vpdhctx, const OSSL_PARAM params[])
 
         EVP_MD_free(pdhctx->kdf_md);
         pdhctx->kdf_md = EVP_MD_fetch(pdhctx->libctx, name, mdprops);
+        if (pdhctx->kdf_md == NULL)
+            return 0;
```

Comment:
ossl_digest_is_allowed handles NULL md gracefully. So this should not be needed as there is another check below. However perhaps somewhat refactoring the code so the duplicate check below is not needed would be even better?

Reply:
I.E., drop the duplicate check below and add return 0 inside the following `if`.